### PR TITLE
SC-1319 - Stronger rules for passwords

### PR DESCRIPTION
**Pull request by ArjunKhetia-NIIT**
SC-1319 - Stronger rules for passwords

### CODE REVIEW

File: src/app/client/src/app/modules/public/module/signup/components/signup/signup.component.ts [hunk 1]
```diff
@@ -36,6 +36,8 @@ export class SignupComponent implements OnInit, OnDestroy, AfterViewInit {
   submitInteractEdata: IInteractEventEdata;
   telemetryCdata: Array<{}>;
   instance: string;
+  isPasswordError = false;
+  passwordErrorMsg: any;
```
Comment (rajeevsathish) [inline src/app/client/src/app/modules/public/module/signup/components/signup/signup.component.ts]: as the message will be a string can we change the object type from any to string

Comment (ArjunKhetia-NIIT) [reply src/app/client/src/app/modules/public/module/signup/components/signup/signup.component.ts]: Done and raised PR.

File: src/app/client/src/app/modules/public/module/signup/components/signup/signup.component.html [hunk 2]
```diff
@@ -49,7 +49,7 @@
                 <i class=" icon">+91-</i>
               </div>
             </div>
-            <label class="ui basic blue error label pt-0" *ngIf="signUpForm.controls.password.touched && signUpForm.controls['password'].errors">{{resourceService.frmelmnts.lbl.validPassword}}</label>
+            <label class="ui basic blue error label pt-0" *ngIf="signUpForm.controls.password.touched && isPasswordError">{{passwordErrorMsg}}</label>
```
Comment (vinukumar-vs) [inline src/app/client/src/app/modules/public/module/signup/components/signup/signup.component.html]: We dont need "isPasswordError" field. Remove this field & relay of "passwordErrorMsg" filed itself. Based on length show the message.

File: src/app/resourcebundles/data/consumption/en.properties [hunk 3]
```diff
@@ -488,6 +488,11 @@ frmelmnts.lbl.userId = User ID
 frmelmnts.lbl.userType = User Type
 frmelmnts.lbl.username = User Name
 frmelmnts.lbl.validPassword = Enter a valid password
+frmelmnts.lbl.charError = Password must contain at least 8 alphanumeric characters.
```
Comment (vinukumar-vs) [inline src/app/resourcebundles/data/consumption/en.properties]: use the below key & follow the same for all keys
frmelmnts.lbl.**password**.lwcsError

**Comment by vinukumar-vs** [comment]
@ArjunKhetia-NIIT resolve the conflicts as well

**Pull request closed without merging**